verilog/tools/ls/vscode/src/download-ls.ts

```diff
@@ -0,0 +1,108 @@
+import { platform, arch } from "os";
+import * as fs from "fs";
+import * as path from "path";
+import { IncomingMessage } from "http";
+import { https as httpsFr } from "follow-redirects";
+import { execSync } from "child_process";
+import * as decompress from "decompress";
+const decompressTargz = require("decompress-targz");
+const TAG = require("../package.json").repository.tag;
+
+
+function checkIfBinaryExists(binaryPath: string) {
+  let whichCommand: string, binaryExists: boolean;
+  if (platform() == "win32") whichCommand = "where";
+  else whichCommand = "which";
+
+  binaryExists = true;
+  try {
+    execSync(`${whichCommand} ${binaryPath}`, { windowsHide: true });
+  } catch {
+    binaryExists = false;
+  }
+
+  return binaryExists;
+}
+
+export async function checkAndDownloadBinaries(
+  binaryPath: string
+): Promise<string> {
+  if (platform() === "darwin") {
+    // There is no static binaries for MacOS -- aborting
+    return binaryPath;
+  }
+  if (checkIfBinaryExists(binaryPath)) {
+    // Language server binary exists -- nothing to do
+    return binaryPath;
+  }
+
+  const pluginDir = path.join(__dirname, "..");
+  const binDir = path.join(pluginDir, "bin");
+  const pluginBinaryPath = path.join(
+    binDir,
+    "verible-verilog-ls" + (platform() === "win32" ? ".exe" : "")
+  );
+  if (checkIfBinaryExists(pluginBinaryPath)) {
+    // Language server binary already downloaded
+    return pluginBinaryPath;
```

Comment:
Is the path dependend on the version of the plug-in ? Otherwise, if we have downloaded it a while ago, it will be stale.
If not, maybe also checking if the binary is older than, say, a month and if so regard it too old.

Reply:
Yes, this path is depended on the version, because downloaded executable (with tag matching the plug-in's version) is saved in the plug-in's directory and VSCode creates separate folder for each versions.

Moreover, updating extension will also update executable -- assuming `verible.path` from settings doesn't point to installed `verible-verilog-ls`

Reply:
VSCode claims to remove old or uninstalled extensions right after restart and when I tested it, old folders was removed a few minutes after restarting app. Thus, I didn't implement any cleanup.